CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+   Keep functions short and simple
+-  [F.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-out-multi)
+   To return multiple 'out' values, prefer returning a ``std::tuple``
+-  [R.1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rr-raii)
+   Manage resources automatically using RAII (this includes
+   ``std::unique_ptr`` & ``std::shared_ptr``)
+-  [ES.11](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-auto)
+   Use ``auto`` to avoid redundant repetition of type names
+-  [ES.20](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-always)
+   Always initialize an object
+-  [ES.23](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-list)
+   Prefer the ``{}`` initializer syntax
+-  [CP.1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#S-concurrency)
+   Assume that your code will run as part of a multi-threaded program
+-  [I.2](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-global)
+   Avoid global variables
+
+## Pull Request Guidelines ##
+Our code contribution guidelines closely follows the model of [GitHub Pull-Requests](https://help.github.com/articles/using-pull-requests).
+The rocWMMA repository follows a workflow which dictates a ``/master`` branch where releases are cut, and a ``/develop`` branch which serves as
+an integration branch for new code.
+
+No changes are allowed to be directly committed to the develop branch of the rocWMMA repository. All authors are required to develop their
+change sets on a separate branch and then create a pull request to merge their changes into the develop branch. When you create a pull
+request, you should target the **develop** branch for integration.
+
+The typical workflow for creating a rocWMMA pull request is as follows:
+
+1. Create and track a rocWMMA fork.
+2. Clone your fork:
+
+    ```bash
+    git clone -b develop https://github.com/<your_fork>/rocWMMA.git .
+    .githooks/install
+    git checkout -b <new_branch>
+    ...
+    git add <new_work>
+    git commit -m "What was changed"
+    git push origin <new_branch>
+    ...
+    ```
+
+3. Create a pull request to the ROCmSoftwarePlatform/rocWMMA develop branch.
+4. Await CI and approval feedback.
+5. Once approved, merge.
+
+**Note**
+You must install GitHooks because there are triggers for Clang formatting in commits. Instructions for formatting
+rocWMMA are included in [Formatting](#formatting).
+
+### Deliverables ###
+
+rocWMMA has a set of required deliverables for every pull request that are as follows.
+
+1. **Test Integration**:
+    - All new functionality introduced to rocWMMA must be accompanied by unit tests. Unit tests should integrate within the existing
+    googletest framework and must have good code coverage. Existing unit tests should be used as a guide and are found in ``test/unit``.
```

Comment:
add: "Be sure to consider rocWMMA's support matrix for datatypes, block sizes and architectures."